include/dali/core/unique_handle.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_UNIQUE_HANDLE_H_
+#define DALI_CORE_UNIQUE_HANDLE_H_
+
+#include <utility>
+#include "dali/core/common.h"
+
+namespace dali {
+
+template <typename HandleType, typename Actual>
+class UniqueHandle {
```

Comment:
Could you describe this? I mean, it's hard to even review, if I don't know what `UniqueHandle` is for

Reply:
Fair point, this class is rather a basic one and, as such, deserves proper docs.

Reply:
Done.